### PR TITLE
vampire update

### PR DESCRIPTION
**Pull request by Nukechickenu64**
# About The Pull Request
makes the coffin heal toxin and burn, toxin because toxin is a serious issue when the vampires go crit cause they get a steady amount of damage, this is compatible with slime species, burn heals too, adds a ressurect action to vampires which they can only use in a coffin and when dead, when bit by a vampire you now drip a tiny amount onto the tile

## Why It's Good For The Game
makes it possible to revive vampires without bullshit

## A Port?
no

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
fix: made the vampire heal toxin in coffins so its alot easier to revive them after being crit for a long time
added: added a ressurect action to vampires that only works in coffins
/:cl:

### CODE REVIEW

**Comment by JerryWester** [comment]
Out of curiosity:
> fix: made the vampire heal toxin in coffins
Does this account for slimes?

File: modular_splurt/code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -470,9 +473,17 @@
 		C.adjustOxyLoss(-4)
 		C.adjustCloneLoss(-4)
 		C.adjustBruteLoss(-0.3)
+		C.adjustFireLoss(-0.3)
+		if(!istype(H.dna.species, /datum/species/jelly)) //checks species
+			C.adjustToxLoss(-5)//heals toxin if not slime
+		else
+			C.adjustToxLoss(5)//heals toxin if slime
```
Comment (JerryWester) [inline modular_splurt/code/datums/traits/neutral.dm:480]: Pfft, nevermind, retract my previous question

**Comment by Nukechickenu64** [comment]
> Out of curiosity:
> > fix: made the vampire heal toxin in coffins
> 
> Does this account for slimes?

Yes

File: modular_splurt/code/datums/traits/neutral.dm [hunk 2]
```diff
@@ -454,11 +454,14 @@
 	if(!H.dna.skin_tone_override)
 		H.skin_tone = "albino"
 	var/datum/action/vbite/B = new
+	var/datum/action/vrevive/R = new
```
Comment (BongaTheProto) [inline modular_splurt/code/datums/traits/neutral.dm:457]: You need to add a removal for this in on_remove like for the vbite one

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

File: modular_splurt/code/datums/traits/neutral.dm [hunk 3]
```diff
@@ -470,19 +473,43 @@
...
+		else
+			H.adjustStaminaLoss(-1,FALSE, FALSE)//this also helps with if someone is stuck in the chapel for way too long, and i tested with a stun sword that stunning for sec is still possible
+		if(rand(0,100) <= 2) //2 percent chance (if it was true randome D:<)
+			to_chat(H, "<span class='warning'>I need blood NOW!!!</span>")
 
 /datum/quirk/vampire/remove()
+	. = ..()
 	var/mob/living/carbon/human/H = quirk_holder
 	var/datum/action/vbite/B = locate() in H.actions
+	var/datum/action/vbite/R = locate() in H.actions
```
Comment (BongaTheProto) [inline modular_splurt/code/datums/traits/neutral.dm:494]:
```suggestion
	var/datum/action/vrevive/R = locate() in H.actions
```

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

File: modular_splurt/code/datums/traits/neutral.dm [hunk 4]
```diff
@@ -454,11 +454,14 @@
...
 		H.skin_tone = "albino"
 	var/datum/action/vbite/B = new
+	var/datum/action/vrevive/R = new
 	B.Grant(H)
+	R.Grant(H)
 
 /datum/quirk/vampire/on_process()
 	. = ..()
 	var/mob/living/carbon/C = quirk_holder
+	var/mob/living/carbon/human/H = quirk_holder
```
Comment (BongaTheProto) [inline modular_splurt/code/datums/traits/neutral.dm:464]: Just remove the C variable and use H for everything

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

File: modular_splurt/code/datums/traits/neutral.dm [hunk 5]
```diff
@@ -479,7 +479,10 @@
 		else
 			C.adjustToxLoss(5)//heals toxin if slime
 		return
-
+	if(H.nutrition =< 0)
+		H.adjustStaminaLoss(3, FALSE, TRUE)
+		if(rand(0,100) =< 10) //10 percent chance (if it was true randome D:<)
```
Comment (JerryWester) [inline modular_splurt/code/datums/traits/neutral.dm]: I think you can just do `rand(10)`

Comment (BongaTheProto) [reply modular_splurt/code/datums/traits/neutral.dm]: > I think you can just do `rand(10)`

You mean `prob(10)`

File: modular_splurt/code/datums/traits/neutral.dm [hunk 6]
```diff
@@ -454,35 +454,61 @@
 	if(!H.dna.skin_tone_override)
 		H.skin_tone = "albino"
 	var/datum/action/vbite/B = new
+	var/datum/action/vrevive/R = new
 	B.Grant(H)
-
+	R.Grant(H)
+species
```
Comment (BongaTheProto) [inline modular_splurt/code/datums/traits/neutral.dm:460]:
```suggestion

```

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

**Comment by JerryWester** [comment]
For being a neutral quirk, this is adding quite a bit of positive traits

**Comment by Nukechickenu64** [comment]
> For being a neutral quirk, this is adding quite a bit of positive traits

might add the ability to remove some of these traits by external forces, like a stake after death removes the ability to resurrect, and remember that ressurecting makes them extremely dizzy and have 0 nutrition which is even worst for them cause theyre stamina gets maxed at 1 when they have 0 nutrition

**Pull request merged**